app/HutchAgent/Services/WorkflowTriggerService.cs

```diff
@@ -19,20 +23,75 @@ public class WorkflowTriggerService : BackgroundService
     _workflowOptions = workflowOptions.Value;
   }
 
-  protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+  /// <summary>
+  /// Parse ROCrate metadata using the ROCrates library
+  /// </summary>
+  /// <param name="jsonFile"> JSON Metadata file</param>
+  /// <returns> ROCrate object </returns>
+  private ROCrate ParseCrate(string jsonFile)
+  {
+    try
+    {
+      // get metadata from manifest
+      var metadataProperties = JsonNode.Parse(jsonFile)?.AsObject();
+      metadataProperties.TryGetPropertyValue("@graph", out var graph);
+
+      // get RootDataset Properties and add them to an ROCrates object
+      var rootDatasetProperties = graph.AsArray().Where(g => g["@id"].ToString() == "./");
+      var datasetRoot = RootDataset.Deserialize(rootDatasetProperties.First().ToString(), _roCrate);
+
+      _roCrate.Add(datasetRoot ?? throw new InvalidOperationException());
+
+      return _roCrate;
+    }
+    catch
+    {
+      throw new Exception("Metadata JSON could not be parsed");
+    }
+  }
+
+  /// <summary>
+  /// Unpack an ROCrate
+  /// </summary>
+  /// <param name="stream"></param>
+  /// <exception cref="FileNotFoundException"></exception>
+  private void UnpackCrate(Stream stream)
   {
-    _logger.LogInformation(
-      "Executing Workflow with WfExS started.");
+    using var archive = new ZipArchive(stream);
+    {
+      // Extract to Directory
+      archive.ExtractToDirectory(_workflowOptions.CrateExtractPath, true);
 
-    await TriggerWfexs();
+      // Validate it is an ROCrate
+      var file = Directory.GetFiles(_workflowOptions.CrateExtractPath, searchPattern: "*metadata.json");
```

Comment:
Change this search pattern to look for a file call `ro-crate-metadata.json`. The spec says that what the file must be called. https://www.researchobject.org/ro-crate/1.1/structure.html#ro-crate-metadata-file-ro-crate-metadatajson